src/main/java/io/github/protocol/prom/PromApiClientImpl.java

```diff
@@ -0,0 +1,212 @@
+package io.github.protocol.prom;
+
+import io.github.protocol.prom.module.VectorResp;
+import io.github.protocol.prom.module.VectorResult;
+import io.github.protocol.prom.util.JacksonUtil;
+import lombok.extern.slf4j.Slf4j;
+import okhttp3.HttpUrl;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.List;
+
+@Slf4j
+public class PromApiClientImpl implements PromApiClient {
+
+    private final String promHttpPrefix;
+
+    private final OkHttpClient client;
+
+    PromApiClientImpl(String host, int port) {
+        this.promHttpPrefix = "http://" + host + ":" + port;
+        OkHttpClient.Builder builder = new OkHttpClient.Builder();
+        builder.connectTimeout(Duration.ofMillis(30_000));
+        this.client = builder.build();
+    }
+
+    // only for testing
+    PromApiClientImpl() {
+        this.promHttpPrefix = null;
+        this.client = null;
+    }
+
+
+    @Override
+    public VectorResp query(String query) throws IOException {
+        HttpUrl parse = HttpUrl.parse(promHttpPrefix + "/api/v1/query");
+        if (parse == null) {
+            throw new IllegalArgumentException("parse url error");
+        }
+        HttpUrl.Builder urlBuilder = parse.newBuilder();
+        urlBuilder.addQueryParameter("query", query);
+        Request request = new Request.Builder()
+                .url(urlBuilder.build())
+                .build();
+        try (Response response = this.client.newCall(request).execute()) {
+            ResponseBody body = response.body();
+            String content = body == null ? "" : body.string();
+            if (!response.isSuccessful()) {
+                log.error("query prometheus failed, response code is {} msg {}", response.code(), content);
+                throw new IOException(String.format("query prometheus failed, error: %s", content));
+            }
+            return JacksonUtil.toObject(content, VectorResp.class);
+        }
+    }
+
+    @Override
+    public Double queryTotalPulsarRateIn() throws IOException {
```

Comment:
```suggestion
    public Double queryPulsarTotalRateIn() throws IOException {
```
I suggent put the **pulsar** behind **Total**. This will make it clearer when we add other components, like add zookeeper. And the other methods